Userland/Libraries/LibWeb/Layout/ReplacedBox.h

```diff
@@ -21,6 +21,15 @@ class ReplacedBox : public Box {
     DOM::Element const& dom_node() const { return verify_cast<DOM::Element>(*Node::dom_node()); }
     DOM::Element& dom_node() { return verify_cast<DOM::Element>(*Node::dom_node()); }
 
+    Optional<CSSPixels> intrinsic_width() const { return m_intrinsic_width; }
+    Optional<CSSPixels> intrinsic_height() const { return m_intrinsic_height; }
+
+    bool has_intrinsic_width() const { return intrinsic_width().has_value(); }
+    bool has_intrinsic_height() const { return intrinsic_height().has_value(); }
+
+    void set_intrinsic_width(Optional<CSSPixels> width) { m_intrinsic_width = width; }
+    void set_intrinsic_height(Optional<CSSPixels> height) { m_intrinsic_height = height; }
```

Comment:
Huh, looks like `m_intrinsic_*` in `ReplacedBox` are not actually used before this change.
Instead of doing this, let's just remove them and instead use the "natural" values in `Box`.
Then we also don't need to check if it's a replaced box at all.

Reply:
Hmm, I don't quite understand what you'd like me to change here. The existing solution did use the "natural" values in `Box` but it wasn't rendering correctly. 

The new code in `FlexFormattingContext` only happens for `ImageBox`es right now, but I feel like looking at the spec, we need to keep track of whether images have a `width` and `height` attribute and change our sizing algorithms accordingly.

Reply:
What I'm saying is: "has intrinsic width" and "has intrinsic height" doesn't really make sense anymore. The spec has renamed these concepts to "natural width" and "natural height" some time ago.

We shouldn't be adding "intrinsic size" to ReplacedBox when its parent class, Box, already has "natural size" :)

Reply:
If I was to remove the "intrinsic width" and "intrinsic height" I'd have nothing to go off of for the added condition on line 653 in `FlexFormatingContext.cpp`. How would you propose I solve this? Or do you think there is a spec bug somewhere?

Reply:
It doesn't seem like the code you changed is part of a formal spec

Reply:
No, it looks like I've misunderstood the spec (which renamed natural height to intrinsic height) but this change fixes up some rendering issues I've seen. I'd love to file a spec bug if it's actually a bug in the spec but I can't be sure of that 😅